Spec/Tests/RealtimeClientConnectionTests.swift

```diff
@@ -2283,95 +2288,52 @@ class RealtimeClientConnectionTests: XCTestCase {
         client.internal.shouldImmediatelyReconnect = false
         defer { client.dispose(); client.close() }
 
-        let ttlHookToken = client.overrideConnectionStateTTL(0.3)
+        let ttlHookToken = client.overrideConnectionStateTTL(4)
         defer { ttlHookToken.remove() }
 
+        var retryCount = 0
+        let maxRetryCount = 8 // RTN14f: reasonable amount within testTimeout
```

Comment:
Perhaps a comment explaining how you calculated this would make this number a bit less magical and give people more confidence to change the test in the future?

Reply:
If we're going to do it this way, can we have an assertion or something that checks that `maxSuspendedRetryCount` is  greater than, say, 2? We definitely want to be sure it's non-zero.

The alternative would be, instead of calculating `maxSuspendedRetryCount` based on { `testTimeout`, `connectionStateTtl`, `options.suspendedRetryTimeout` }, to choose the following:

- how long do we want the test to run for? (which then means that the test doesn't need to run for `testTimeout` (20 seconds) which is a long time)
- how many retries do we want to observe in order to get confidence that everything is working properly?

and then to calculate { `connectionStateTtl`, `options.suspendedRetryTimeout` } based on those. I think that this would make the test more understandable.

Reply:
I've chosen 3 times, I think that's enough 🧐 https://github.com/ably/ably-cocoa/pull/1520/commits/564498baacfcf91dbc6ac805ad645ee788f81118